awsshell/utils.py

```diff
@@ -123,12 +125,19 @@ def file_exists(self, filename):
 
 def force_unicode(obj, encoding='utf8'):
     """Recursively search lists and dicts for strings to encode as unicode."""
+    def attempt_decode(string):
```

Comment:
It is a little strange to define the function each time ``force_unicode`` is called. Typically I would extract this out to the module level or not even create a function in the first place since it is only used once.